flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -224,23 +226,26 @@ private boolean isFinalState(ComputationState state) {
 
 	/**
 	 * Prunes states assuming there will be no events with timestamp <b>lower</b> than the given one.
-	 * It cleares the sharedBuffer and also emits all timed out partial matches.
+	 * It clears the sharedBuffer and also emits all timed out partial matches.
 	 *
 	 * @param sharedBuffer the SharedBuffer object that we need to work upon while processing
 	 * @param nfaState     The NFAState object that we need to affect while processing
 	 * @param timestamp    timestamp that indicates that there will be no more events with lower timestamp
 	 * @return all timed outed partial matches
 	 * @throws Exception Thrown if the system cannot access the state.
 	 */
-	public Collection<Tuple2<Map<String, List<T>>, Long>> advanceTime(
+	public Tuple2<Collection<Tuple2<Map<String, List<T>>, Long>>, Collection<Map<String, List<T>>>> advanceTime(
 			final SharedBuffer<T> sharedBuffer,
 			final NFAState nfaState,
-			final long timestamp) throws Exception {
+			final long timestamp,
+			final AfterMatchSkipStrategy afterMatchSkipStrategy) throws Exception {
```

Comment:
please add parameter explication for new parameter `afterMatchSkipStrategy ` of the method.